### PR TITLE
Gd sing in modal updates

### PR DESCRIPTION
**Pull request by redpixr**
- Modify modal layout, add google & fb buttons
- Add refs to buttons
- Add google sign in buttons
- Add button click handler logic for auth
- Add google & fb login requests
- Fix the donation start date text being null
- Add catch statements to popup sign in calls
- Fix the token being undefined issue after sign out
- Fix responsive issues on sign in modal
- Initial Google sign in error handling
- Implement credential exists handling

### CODE REVIEW

File: static/js/authController.ts [hunk 1]
```diff
@@ -122,6 +122,20 @@ export class AuthController {
     return this.token ? this.token : null;
   }
 
+  public signInWithGoogle(): void {
+    const googleAuth = new firebase.auth.GoogleAuthProvider();
+    firebase.auth().signInWithPopup(googleAuth).catch((error) => {
+      console.log('Authentication was unsuccessful!');
```
Comment (bookjunkie93) [inline static/js/authController.ts:128]: Before this gets committed to master we should look through the possible auth errors and make sure we're handling common ones

File: static/js/authController.ts [hunk 2]
```diff
@@ -122,6 +122,20 @@ export class AuthController {
...
+    });
+  }
+
+  public signInWithFacebook(): void {
+    const facebookAuth = new firebase.auth.FacebookAuthProvider();
+    firebase.auth().signInWithPopup(facebookAuth).catch((error) => {
+      console.log('Authentication was unsuccessful!');
```
Comment (bookjunkie93) [inline static/js/authController.ts:135]: same as above

File: static/js/signInModal.ts [hunk 3]
```diff
@@ -131,7 +149,7 @@ export class SignInModal extends Modal {
       } else {
         this.donorInfoText?.classList.remove(this.hiddenClass);
         this.donorInfoText!.innerHTML = data.displayText;
-        this.instructionText?.classList.add(this.hiddenClass);
+        // this.instructionText?.classList.add(this.hiddenClass);
```
Comment (bookjunkie93) [inline static/js/signInModal.ts:152]: If we're no longer going to use this statement it can be deleted

File: views/index.pug [hunk 4]
```diff
@@ -323,34 +323,60 @@
...
-          p.has-text-info.is-hidden#new-donor-info-text
+          p.has-text-info.is-hidden#new-donor-info-text(style='text-align: center;')
             span.icon.is-small.is-left
               i.fas.fa-info-circle
             span#new-donor-info-content Please enter a valid email address
             br
             br
-          p.has-text-grey#modal-instruction-text
-            span We’ll email you a magic code for a password-free sign in.
+          //- p.has-text-grey#modal-instruction-text
```
Comment (bookjunkie93) [inline views/index.pug:336]: If we're no longer going to use this text we can delete it

**Review by bookjunkie93 — APPROVED** [review]
Minor comments to keep in mind as we move forward. This is good for adding to the QA server, but we need some cleanup and error handling before pushing to master

File: static/js/authController.ts [hunk 5]
```diff
@@ -122,6 +122,63 @@ export class AuthController {
...
+    firebase.auth().signInWithPopup(facebookAuth).catch((error) => {
+      if (error.code === 'auth/account-exists-with-different-credential') {
+        let pendingCredential = error.credential;
+        let email = error.email;
+
+        firebase.auth().fetchSignInMethodsForEmail(email).then((methods: any) => {
+          if (methods.length > 0) {
+            window.alert(`Your auth methods${methods.length}: ${methods[0]}`);
+            if (methods[0] === "google.com") {
+              if (window.confirm('You have already authorized using a Facebook account. Click yes if you\'d like to sign in with Facebook and click no if you wish to sign in with a Google account and link to the existing one.')) {
```
Comment (bookjunkie93) [inline static/js/authController.ts:165]: this can be extracted into a config file

File: static/js/authController.ts [hunk 6]
```diff
@@ -122,6 +122,63 @@ export class AuthController {
...
+    firebase.auth().signInWithPopup(googleAuth).catch((error) => {
+      if (error.code === 'auth/account-exists-with-different-credential') {
+        let pendingCredential = error.credential;
+        let email = error.email;
+
+        firebase.auth().fetchSignInMethodsForEmail(email).then((methods: any) => {
+          if (methods.length > 0) {
+            window.alert(`Your auth methods${methods.length}: ${methods[0]}`);
+            if (methods[0] === "facebook.com") {
+              if (window.confirm('You have already authorized using a Facebook account. Click yes if you\'d like to sign in with Facebook and click no if you wish to sign in with a Google account and link to the existing one.')) {
```
Comment (bookjunkie93) [inline static/js/authController.ts:137]: This can be extracted into a config file

File: static/js/authController.ts (hunk 5 above)
Comment (bookjunkie93) [reply static/js/authController.ts:165]: There's also probably an argument for extracting the `fetchSignInMethodsForEmail` call into its own method with params

**Review by bookjunkie93 — APPROVED** [review]

File: static/js/authController.ts (hunk 2 above)
Comment (burrage) [reply static/js/authController.ts]: @redpixr

**Review by bookjunkie93 — APPROVED** [review]

**Pull request merged**